internal/bufio/bufio.go

```diff
@@ -0,0 +1,205 @@
+// Package bufio provides specialised packet buffers.
+package bufio
```

Comment:
Does it have any connection to https://pkg.go.dev/bufio ? Be careful when naming a package the same as a standard library package, because the caller will likely not rename it, and therefore it might be confusing for readers that would mistake it for the std lib 😬

Reply:
Good point. I don't really have a good name for it, so picked `buffr.BufferedReader`. WDYT?

Reply:
Would make it a `buffr.Reader` then to avoid the repetition

(Ah also I forgot to mention it but we use American spelling so it should be `specialized` instead of specialised)

Reply:
I like `buffr.Reader`. 

Specialised isn't there anymore, it's now `// Package buffr provides a buffered reader.`.